test-cypress/integration/repository/repositories.spec.js

```diff
@@ -0,0 +1,404 @@
+        cy.get('.ot-splash').should('not.be.visible');
+
+        getRepositoriesDropdown().should('be.visible').and('not.be.disabled');
+        getCreateRepositoryButton().should('be.visible').and('not.be.disabled');
+    }
+
+    it('should allow creation of repositories with default settings', () => {
+        // There should be a default repository location
+        getLocationsList()
+            .should('have.length', 1)
+            .and('contain', 'Local');
+
+        getRepositoriesDropdown()
+            .find('.no-selected-repository')
+            .should('be.visible');
+
+        createRepository();
+
+        cy.url().should('include', '/repository/create');
+
+        // Create a repository by supplying only an identifier
+        getRepositoryCreateForm().should('be.visible');
+        getRepositoryIdField()
+            .should('have.value', '')
+            .type(repositoryId)
+            .should('have.value', repositoryId);
+        saveRepository();
+
+        cy.url().should((url) => {
+            expect(url.endsWith('/repository')).to.equal(true);
+        });
+
+        // Check the repo is present in the list of repos and we are not yet connected to it
+        getRepositoryFromList(repositoryId)
+            .should('be.visible')
+            .find('.icon-connection-off')
+            .should('be.visible');
+
+        // Verify it's configuration can be downloaded
+        getRepositoryFromList(repositoryId)
+            .find('.repository-actions .download-repository-config')
+            .should('be.visible')
+            .and('not.be.disabled');
+
+        // Connect to the repository via the menu
+        getRepositoriesDropdown().click().within(() => {
+            cy.get('.no-selected-repository').should('be.visible');
+
+            // The dropdown should contain the newly created repository
+            cy.get('.dropdown-menu .dropdown-item')
+                .contains(repositoryId)
+                .scrollIntoView()
+                .click();
+
+            // Should visualize the selected repo
+            cy.get('.no-selected-repository').should('not.be.visible');
+            cy.get('.active-repository')
+                .should('be.visible')
+                .and('contain', repositoryId);
+        });
+
+        // The repo should be connected after selecting it from the menu
+        getRepositoryFromList(repositoryId)
+            .find('.icon-connection-on')
+            .should('be.visible');
+
+        // And it should not be present in the dropdown items
+        getRepositoriesDropdown()
+            .click()
+            .find('.dropdown-menu .dropdown-item')
+            .should('not.contain', repositoryId);
+    });
+
+    it('should disallow creation of repositories without mandatory settings', () => {
+        createRepository();
+        saveRepository();
+
+        getRepositoryCreateForm().should('be.visible');
+        getRepositoryIdField().should('have.class', 'ng-invalid');
```

Comment:
Maybe its better to check for the placeholder `This field is required` instead relying of the angular class.

Reply:
The problem is that according to Cypress the DOM is "visible" and "actionable" when in fact the snapshots shows it's not yet rendered or properly rendered (due to the fact animations are not finished I think). So that's why I want to wait until the animation is finished.

Reply:
Updated to check for placeholder's text